test/App/Fossa/LernieSpec.hs

```diff
@@ -391,10 +402,31 @@ spec = do
 
     it' "should merge the config from fossa.yml and the org" $ do
       GetOrganization `alwaysReturns` Fixtures.organization{orgCustomLicenseScanConfigs = [secondCustomLicenseGrepEntry]}
-      result <- ignoreDebug . withoutTelemetry $ analyzeWithLernieWithOrgInfo scanDir grepOptions
+      result <- ignoreDebug . withoutTelemetry $ analyzeWithLernieWithOrgInfo scanDir grepOptions Nothing
       case result of
         Nothing -> expectationFailure' "analyzeWithLernie should not return Nothing"
         Just res -> do
           -- Just assert that we find matches for "Confidential" (from the org API) and "Proprietary License" (from grepOptions)
           let matchNames = lernieMatchDataName <$> concatMap lernieMatchMatches (lernieResultsCustomLicenses res)
           sort (nub matchNames) `shouldBe'` ["Confidential", "Proprietary License"]
+
+    it' "should handle Nothing licenseScanPathFilters without crashing" $ do
+      result <- ignoreDebug . withoutTelemetry $ analyzeWithLernie scanDir Nothing grepOptions Nothing
+      case result of
+        Nothing -> expectationFailure' "analyzeWithLernie should not return Nothing"
+        Just _ -> pure ()
+
+    it' "should apply licenseScanPathFilters correctly" $ do
+      let filters =
+            LicenseScanPathFilters
+              { licenseScanPathFiltersOnly = [GlobFilter "**/*one.txt"]
+              , licenseScanPathFiltersExclude = [GlobFilter "**/*something.txt"]
```

Comment:
[nit] This isn't testing the exclude filter, as the `only` filter is already filtering out something.txt. It's probably worth adding a test of just the exclude filter too.